tools/esp-api/espshell.cpp

```diff
@@ -52,33 +52,32 @@ void EspShell::usage()
     "    esp-api describe WsWorkunits WUDelete           Describes the request and response structure of the 'WUDelete' method.\n"
     "\n"
     "    esp-api test WsWorkunits WUDelete --username <username> --password <password> "
-    "--server http://127.0.0.1 --port 8010 --reqStr '{\"WUDeleteRequest\":{\"Wuids\":{\"Item\":[\"test\"]},\"BlockTillFinishTimer\":0}}'\n"
+    "--server http://127.0.0.1 --port 8010 --query ver_=1.2 --reqStr '{\"WUDeleteRequest\":{\"Wuids\":{\"Item\":[\"test\"]},\"BlockTillFinishTimer\":0}}'\n"
     "Note:\n"
-    "    To autocomplete the service and method names, use the 'Tab' key after typing part of the command.\n\n"
+    "    To autocomplete the service and method names, use the 'Tab' key after typing the command.\n\n"
 );
 }
 
 string EspShell::buildUrl(const char* server, const char* port, const char* query, const char* resType, const char* serviceName, const char* methodName)
 {
     std::ostringstream url;
-    url << server << ":" << port << "/" << serviceName << "/" << methodName << resType;
-    return url.str();
-}
-
-int EspShell::sendRequest(const char* serviceName, const char* methodName, const char* formArgs, const char* resType, const char* reqType, const char* server
-                            ,const char* port, const char* query)
-{
-    string url = buildUrl(server, port, query, resType, serviceName, methodName);
-    if(hasUsername && hasPassword)
+    if(isEmptyString(query))
     {
-        EspService myServ(serviceName, methodName, formArgs, resType, reqType, url.c_str(), username.str(), password.str());
-        return myServ.sendRequest();
+        url << server << ":" << port << "/" << serviceName << "/" << methodName << resType;
     }
     else
     {
-        EspService myServ(serviceName, methodName, formArgs, resType, reqType, url.c_str());
-        return myServ.sendRequest();
+        url << server << ":" << port << "/" << serviceName << "/" << methodName << resType << "?" << query;
```

Comment:
Be aware of redundant code. In this case, `url` can be set unconditionally up to and including `resType`. Then `"?" << query` can be added conditionally.

Reply:
This may be a good candidate to defer to a 'cleanup' ticket if you don't have time to do it now. If so please post a link to the ticket in a comment.